'SpecReporter' is not assignable to type 'Reporter | CustomReporter'

# Issue guidelines

## Bug report

Check with "x":

- [x] I have this issue with the latest release of [jasmine-spec-reporter](https://github.com/bcaudan/jasmine-spec-reporter/releases) 
- [x] I have this issue with the latest release of [jasmine](https://github.com/jasmine/jasmine/releases) 
- [ ] I can reproduce this issue with one of the [examples](https://github.com/bcaudan/jasmine-spec-reporter/tree/master/examples)

**Current behavior**

I'm encountering a type error when configuring the spec reporter, probably related to the latest `@types/jasmine`package:

```
index.ts(91,27): error TS2345: Argument of type 'SpecReporter' is not assignable to parameter of type 'Reporter | CustomReporter'.
  Type 'SpecReporter' is not assignable to type 'CustomReporter'.
    Types of property 'suiteStarted' are incompatible.
      Type '(result: import("node_modules/jasmine-spec-reporter/built/spec-reporter").CustomReporterResult) => void' is not assignable to type '(result: jasmine.CustomReporterResult) => void'.
        Types of parameters 'result' and 'result' are incompatible.
          Type 'jasmine.CustomReporterResult' is not assignable to type 'import("node_modules/jasmine-spec-reporter/built/spec-reporter").CustomReporterResult'.
            Types of property 'duration' are incompatible.
              Type 'number | null' is not assignable to type 'string | undefined'.
                Type 'null' is not assignable to type 'string | undefined'.
```

**Expected behavior**

No type error, assignment works as expected.

**Minimal reproduction of the problem with instructions**

Package versions:

- `@types/jasmine`: 3.6.6
- `jasmine`: 3.6.4
- `jasmine-spec-reporter`: 6.0.0

``` ts
const jasmine = new Jasmine({})
jasmine.env.clearReporters()
jasmine.env.addReporter(new SpecReporter({
  spec: { displayStacktrace: StacktraceOption.RAW }
}))
```

**Possible fix**

Change the typing of `duration` to `number | null` here:

https://github.com/bcaudan/jasmine-spec-reporter/blob/8d7bd5f78a640fc2c04dc6397c47ca18b806ac97/src/spec-reporter.ts#L20-L22

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,29 @@
+# 7.0.0
+
+## Bugfix
+
+* 'SpecReporter' is not assignable to type 'Reporter | CustomReporter' [#588](https://github.com/bcaudan/jasmine-spec-reporter/pull/588)
+
+## Breaking change
+
+Update signature of `CustomReporterResult` to fix collision with new jasmine properties
+
+**Before:**
+```ts
+export interface CustomReporterResult extends jasmine.CustomReporterResult {
+  duration?: string;
+}
+```
+
+**Now:**
+```ts
+export interface CustomReporterResult extends jasmine.CustomReporterResult {
+  _jsr?: {
+    formattedDuration?: string;
+  };
+}
+```
+
 # 6.0.0
 
 ## Feature

--- a/package.json
+++ b/package.json
@@ -47,7 +47,7 @@
     "colors": "1.4.0"
   },
   "devDependencies": {
-    "@types/jasmine": "3.6.2",
+    "@types/jasmine": "3.6.9",
     "@types/node": "14.14.37",
     "codecov": "3.8.1",
     "diff": "5.0.0",

--- a/src/display/execution-display.ts
+++ b/src/display/execution-display.ts
@@ -10,7 +10,7 @@ export class ExecutionDisplay {
         let isDisplayed = false;
         processors.forEach((processor: DisplayProcessor) => {
             const log = "foo";
-            const result = processor.displaySpecStarted({id: "bar", description: "bar", fullName: "bar"}, log);
+            const result = processor.displaySpecStarted({id: "bar", description: "bar", fullName: "bar", duration: null, properties: null}, log);
             isDisplayed = isDisplayed || result !== log;
         });
         return isDisplayed;
@@ -120,6 +120,8 @@ export class ExecutionDisplay {
                 description: name,
                 fullName: name,
                 id: name,
+                properties: null,
+                duration: null
             };
             this.suiteHierarchy.push(topLevelSuite);
             this.suiteHierarchyDisplayed.push(topLevelSuite);

--- a/src/execution-metrics.ts
+++ b/src/execution-metrics.ts
@@ -42,7 +42,9 @@ export class ExecutionMetrics {
     }
 
     public stopSpec(result: CustomReporterResult): void {
-        result.duration = this.formatDuration((new Date()).getTime() - this.specStartTime);
+        result._jsr = {
+            formattedDuration: this.formatDuration((new Date()).getTime() - this.specStartTime)
+        };
     }
 
     private formatDuration(durationInMs: number): string {

--- a/src/processors/spec-durations-processor.ts
+++ b/src/processors/spec-durations-processor.ts
@@ -3,7 +3,7 @@ import {CustomReporterResult} from "../spec-reporter";
 
 export class SpecDurationsProcessor extends DisplayProcessor {
     private static displayDuration(spec: CustomReporterResult, log: string): string {
-        return `${log} (${spec.duration})`;
+        return `${log} (${spec._jsr && spec._jsr.formattedDuration})`;
     }
 
     public displaySuccessfulSpec(spec: CustomReporterResult, log: string): string {

--- a/src/spec-reporter.ts
+++ b/src/spec-reporter.ts
@@ -18,7 +18,9 @@ import SuiteInfo = jasmine.SuiteInfo;
 import RunDetails = jasmine.RunDetails;
 
 export interface CustomReporterResult extends jasmine.CustomReporterResult {
-    duration?: string;
+    _jsr?: {
+        formattedDuration?: string;
+    };
 }
 
 export interface ExecutedSpecs {
@@ -136,6 +138,8 @@ export class SpecReporter implements CustomReporter {
             }),
             fullName: "Non-spec failure",
             id: "Non-spec failure",
+            duration: null,
+            properties: null
         };
     }
 

